File: FrontOne.Domain/Entities/Usuario.cs
namespace FrontOne.Domain.Entities;

public class Usuario
{
    public int Id { get; set; }
    public string NombreUsuario { get; set; } = string.Empty;
    public string NombreCompleto { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string PasswordEncriptado { get; set; } = string.Empty;
    public string? PasswordHash { get; set; }
    public bool Activo { get; set; }
    public DateTime FechaCreacion { get; set; }
    public int IntentosFallidos { get; set; }
    public DateTime? BloqueadoHasta { get; set; }
}

File: FrontOne.Android/domain/src/main/kotlin/com/frontone/android/domain/model/Usuario.kt
package com.frontone.android.domain.model

/** Equivalente a UsuarioDto.cs (FrontOne C#) — sin el password, nunca sale del login. */
data class Usuario(
    val id: Int,
    val nombreUsuario: String,
    val nombreCompleto: String,
    val email: String?,
    val activo: Boolean
)

/**
 * Proyección interna usada solo dentro de LoginUseCase — incluye el password
 * cifrado tal como viene de Seguridad.Usuario. Nunca se expone fuera del caso de uso
 * (ResultadoLogin.Exitoso solo entrega [Usuario], sin este campo).
 */
data class UsuarioAutenticacion(
    val id: Int,
    val nombreUsuario: String,
    val nombreCompleto: String,
    val email: String?,
    val activo: Boolean,
    val passwordEncriptado: String
)
